packages/quantic/force-app/main/default/lwc/quanticCaseClassification/__tests__/quanticCaseClassification.test.js

```diff
@@ -0,0 +1,848 @@
+          });
+          mockGetPicklistValuesByRecordTypeWire();
+
+          await flushPromises();
+
+          const suggestions = element.shadowRoot.querySelectorAll(
+            selectors.caseClassificationSuggestion
+          );
+
+          expect(suggestions.length).toBe(exampleMaxSuggestions);
+          for (let index = 0; index < exampleMaxSuggestions; index++) {
+            expect(suggestions[index].textContent).toBe(
+              exampleCaseClassificationSuggestions[index].value
+            );
+          }
+        });
+
+        it('should automatically select the first case classification suggestion', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            maxSuggestions: exampleMaxSuggestions,
+          });
+          mockGetPicklistValuesByRecordTypeWire();
+
+          await flushPromises();
+
+          const suggestionInputs = element.shadowRoot.querySelectorAll(
+            selectors.caseClassificationSuggestionInput
+          );
+
+          expect(functionsMocks.update).toHaveBeenCalledTimes(1);
+          expect(functionsMocks.update).toHaveBeenCalledWith(
+            exampleCaseClassificationSuggestions[0].value,
+            {caseClassifications: false, documentSuggestions: false},
+            true
+          );
+          expect(suggestionInputs[0].checked).toBe(true);
+        });
+
+        describe('when a case classification suggestion is clicked', () => {
+          it('should call the update method of the case field controller with the right parameter', async () => {
+            const automaticallySelectedSuggestionIndex = 0;
+            const exampleSelectedSuggestionIndex = 1;
+            const element = createTestComponent({
+              ...defaultOptions,
+              maxSuggestions: exampleMaxSuggestions,
+            });
+            mockGetPicklistValuesByRecordTypeWire();
+
+            await flushPromises();
+
+            expect(functionsMocks.update).toHaveBeenCalledTimes(1);
+            expect(functionsMocks.update).toHaveBeenCalledWith(
+              exampleCaseClassificationSuggestions[
+                automaticallySelectedSuggestionIndex
+              ].value,
+              {caseClassifications: false, documentSuggestions: false},
+              true
+            );
+
+            const suggestionInputs = element.shadowRoot.querySelectorAll(
+              selectors.caseClassificationSuggestionInput
+            );
+
+            suggestionInputs[exampleSelectedSuggestionIndex].click();
+            await flushPromises();
+
+            expect(functionsMocks.update).toHaveBeenCalledTimes(2);
+            expect(functionsMocks.update).toHaveBeenCalledWith(
+              exampleCaseClassificationSuggestions[
+                exampleSelectedSuggestionIndex
+              ].value,
+              {caseClassifications: false, documentSuggestions: false},
+              undefined
```

Comment:
What's that undefined? Are we specifically sending `undefined` for the `autoSelection`? The parameter is optional no? shouldn't just be not there instead?

Reply:
yes, basically we are always sending `autoSelection` even when it is undefined to avoid having a code like this: 
```
      if (autoSelection) {
        this.field.update(value, updatesToFetch, autoSelection);
      } else {
        this.field.update(value, updatesToFetch);
      }
```
while just 
```
this.field.update(value, updatesToFetch, autoSelection);
```
yield the exact same result anyway.

Reply:
Seems weird?